pkg/controller/machine/machine_controller_suite_test.go

```diff
@@ -61,14 +61,17 @@ func SetupTestReconcile(inner reconcile.Reconciler) (reconcile.Reconciler, chan
 }
 
 // StartTestManager adds recFn
-func StartTestManager(mgr manager.Manager, t *testing.T) chan struct{} {
+func StartTestManager(mgr manager.Manager, t *testing.T) (chan struct{}, chan error) {
 	t.Helper()
 
 	stop := make(chan struct{})
+	errs := make(chan error, 1)
+
 	go func() {
 		if err := mgr.Start(stop); err != nil {
-			t.Fatalf("error starting test manager: %v", err)
+			errs <- err
 		}
```

Comment:
I might suggest you just put this on the channel anyway, doesn't matter if it's a nil value or not, we can check that where it's being read. That way we can also guarantee that we check the error, as currently the select _might_ not catch it in time

So it would look like here:
```
errs <- mgr.Start(stop)
```

and then
```
defer func() {
  if err := <-errChan; err != nil {
    t.Fatalf("error starting test manager: %v", err)
  }
}
```